internal/types/document_validation.go

```diff
@@ -50,6 +52,14 @@ func (d *Document) ValidateData() error {
 		if strings.Contains(key, "$") {
 			return newValidationError(fmt.Errorf("invalid key: %q (key must not contain $)", key))
 		}
+
+		if key == "_id" {
+			idPresent = true
+		}
+	}
+
+	if !idPresent {
+		return newValidationError(fmt.Errorf("invalid document: (document must contain '_id' field)"))
```

Comment:
```suggestion
		return newValidationError(fmt.Errorf("invalid document: document must contain '_id' field"))
```

I think, extra `(` are not needed in this case :)

Reply:
The function is using the same formatting for every other error:
```go
// Tests for this case are in `dance`.
if !utf8.ValidString(key) {
	return newValidationError(fmt.Errorf("invalid key: %q (not a valid UTF-8 string)", key))
}

// Tests for this case are in `dance`.
if strings.Contains(key, "$") {
	return newValidationError(fmt.Errorf("invalid key: %q (key must not contain $)", key))
}
```

Reply:
You're right, nice catch!